src/sidebar/services/test/annotations-exporter-test.js

```diff
@@ -256,4 +258,115 @@ ${isoDate},http://example.com,My group,Annotation,,John Doe,Annotation text,"tag
       );
     });
   });
+
+  describe('buildHTMLExportContent', () => {
+    it('throws error when empty list of annotations is provided', () => {
+      assert.throws(
+        () => exporter.buildHTMLExportContent([]),
+        'No annotations to export',
+      );
+    });
+
+    it('generates HTML content with expected annotations', () => {
+      const isoDate = baseAnnotation.created;
+      const annotations = [
+        {
+          ...baseAnnotation,
+          user: 'acct:jane@localhost',
+          tags: ['foo', 'bar'],
+        },
+        {
+          ...baseAnnotation,
+          ...newReply(),
+          target: targetWithSelectors(
+            quoteSelector('includes <p>HTML</p> tags'),
+            pageSelector(23),
+          ),
+        },
+        {
+          ...baseAnnotation,
+          tags: [],
+          target: targetWithSelectors(pageSelector('iii')),
+        },
+      ];
+
+      const result = exporter.buildHTMLExportContent(annotations, {
+        groupName,
+        now,
+      });
+
+      assert.equal(
+        result,
+        `<header>
+\t<p>
+\t\t<time datetime="${isoDate}">${isoDate}</time>
+\t</p>
+\t<p>
+\t\t<strong>A special document</strong>
+\t</p>
+\t<p>
+\t\t<a
+\t\t\thref="http://example.com"
+\t\t\ttarget="_blank"
+\t\t\trel="noopener noreferrer"
+\t\t>
+\t\t\thttp://example.com
+\t\t</a>
+\t</p>
+\t<dl>
+\t\t<dt>Group</dt>
+\t\t<dd>My group</dd>
+\t\t<dt>Total users</dt>
+\t\t<dd>2</dd>
+\t\t<dt>Users</dt>
+\t\t<dd>jane, bill</dd>
+\t\t<dt>Total annotations</dt>
+\t\t<dd>3</dd>
+\t\t<dt>Total replies</dt>
+\t\t<dd>1</dd>
+\t</dl>
+</header>
+<hr />
+<section>
+\t<article>
+\t\t<strong>Annotation 1:</strong>
+\t\t<p>${isoDate}</p>
+\t\t<p>Comment: Annotation text</p>
+\t\t<p>jane</p>
+\t\t<p>
+\t\t\tQuote: 
+\t\t\t<blockquote></blockquote>
+\t\t</p>
+\t\t<p>Tags: foo, bar</p>
+\t</article>
+\t<br />
+\t<article>
+\t\t<strong>Annotation 2:</strong>
+\t\t<p>${isoDate}</p>
+\t\t<p>Comment: Annotation text</p>
+\t\t<p>bill</p>
+\t\t<p>
+\t\t\tQuote: 
+\t\t\t<blockquote>includes &lt;p>HTML&lt;/p> tags</blockquote>
+\t\t</p>
+\t\t<p>Tags: tag_1, tag_2</p>
+\t\t<p>Page: 23</p>
+\t</article>
+\t<br />
+\t<article>
+\t\t<strong>Annotation 3:</strong>
+\t\t<p>${isoDate}</p>
+\t\t<p>Comment: Annotation text</p>
+\t\t<p>bill</p>
+\t\t<p>
+\t\t\tQuote: 
+\t\t\t<blockquote></blockquote>
+\t\t</p>
+\t\t<p>Page: iii</p>
+\t</article>
+\t<br />
```

Comment:
My IDE insists in reformatting tabs into spaces every time I save this file.

I thought others could have the same problem, so I "hardcoded" tabs as `\t` to prevent that in the future.